### PR TITLE
Retry deleting failed attempt output files

### PR DESCRIPTION
**Pull request by losipiuk**
No RN/Docs

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java [hunk 1]
```diff
@@ -134,6 +136,10 @@
     private static final int PARTITION_COMMIT_BATCH_SIZE = 8;
     private static final Pattern DELTA_DIRECTORY_MATCHER = Pattern.compile("(delete_)?delta_[\\d]+_[\\d]+_[\\d]+$");
 
+    private static final RetryDriver DELETE_RETRY = RetryDriver.retry()
+            .maxAttempts(3)
+            .exponentialBackoff(new Duration(1, SECONDS), new Duration(1, SECONDS), new Duration(10, SECONDS), 2.0);
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java:141]: Any reason for this particular setup?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java [hunk 2]
```diff
@@ -3663,8 +3669,12 @@ public static void cleanExtraOutputFiles(HdfsEnvironment hdfsEnvironment, HdfsCo
             Iterator<String> filesToDeleteIterator = filesToDelete.iterator();
             while (filesToDeleteIterator.hasNext()) {
                 String fileName = filesToDeleteIterator.next();
-                log.debug("Deleting failed attempt file %s/%s for query %s", path, fileName, queryId);
-                checkedDelete(fileSystem, new Path(path, fileName), false);
+                String filePath = path + "/" + fileName;
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java:3672]: You could use `new Path(path, fileName)` here, so you have one variable, not two

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java [hunk 3]
```diff
@@ -3674,7 +3684,7 @@ public static void cleanExtraOutputFiles(HdfsEnvironment hdfsEnvironment, HdfsCo
                 log.info("Deleted failed attempt files %s from %s for query %s", deletedFiles, path, queryId);
             }
         }
-        catch (IOException e) {
+        catch (Exception e) {
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java:3687]: interrupt if interrupted.

**Review by findepi — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java (hunk 1 above)
Comment (losipiuk) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java:141]: It just felt default bail out after 10 retries/30s is an overkill.

**Comment by losipiuk** [comment]
CI: https://github.com/trinodb/trino/issues/12950

**Pull request merged**